core/server/src/main/java/alluxio/worker/block/evictor/LIRSEvictor.java

```diff
@@ -0,0 +1,527 @@
+   * called.
+   */
+  private Map<Long, Long> mBlockIdToSize = new ConcurrentHashMap<Long, Long>();
+  /**
+   * Record the information of all LIR blocks. Pair<BlockStoreLocation, Long> identifies a block
+   * because one block may contain more than one ghost copies on all tiers. One ghost block is
+   * used to mark the block removed from some tier but it may be referenced again in the future.
+   */
+  private Map<Pair<BlockStoreLocation, Long>, BlockType> mLIRCache =
+      Collections.synchronizedMap(new LinkedHashMap<Pair<BlockStoreLocation, Long>, BlockType>(
+          LINKED_HASH_MAP_INIT_CAPACITY, LINKED_HASH_MAP_INIT_LOAD_FACTOR,
+          LINKED_HASH_MAP_ACCESS_ORDERED));
+  // Record the information of all HIR blocks.
+  private Map<Pair<BlockStoreLocation, Long>, BlockType> mHIRCache =
+      Collections.synchronizedMap(new LinkedHashMap<Pair<BlockStoreLocation, Long>, BlockType>(
+          LINKED_HASH_MAP_INIT_CAPACITY, LINKED_HASH_MAP_INIT_LOAD_FACTOR,
+          LINKED_HASH_MAP_ACCESS_ORDERED));
+
+  /**
+   * Create a new instance of {@link LIRSEvictor}.
+   *
+   * @param view a view of block metadata information
+   * @param allocator an allocation policy
+   */
+  public LIRSEvictor(BlockMetadataManagerView view, Allocator allocator) {
+    super(view, allocator);
+    mConfiguration = WorkerContext.getConf();
+    mHIRPercent = mConfiguration.getDouble(Constants.WORKER_EVICTOR_LIRS_HIR_PERCENT);
+    Preconditions.checkArgument(mHIRPercent >= 0 && mHIRPercent <= 1,
+        "HIR percent should be larger than 0 and less than 1");
+    mLIRPercent = 1.0 - mHIRPercent;
+
+    for (StorageTierView tier : view.getTierViews()) {
+      for (StorageDirView dir : tier.getDirViews()) {
+        long lirBytes = 0;
+        long hirBytes = 0;
+        long totalBytes = dir.getAvailableBytes() + dir.getEvitableBytes();
+        double lirLimitBytes = mLIRPercent * totalBytes;
+        BlockStoreLocation location =
+            new BlockStoreLocation(tier.getTierViewAlias(), dir.getDirViewIndex());
+        for (BlockMeta blockMeta : dir.getEvictableBlocks()) {
+          long blockId = blockMeta.getBlockId();
+          long blocksize = blockMeta.getBlockSize();
+          Pair<BlockStoreLocation, Long> key =
+              new Pair<BlockStoreLocation, Long>(location, blockId);
+          // If the size of LIR blocks doesn't exceed the limit, put it into HIR cache.
+          if (lirBytes + blocksize <= lirLimitBytes) {
+            // put into LIR cache.
+            mLIRCache.put(key, BlockType.LIR_RESIDENT);
+            lirBytes += blocksize;
+          } else {
+            // put into HIR cache.
+            mLIRCache.put(key, BlockType.HIR_RESIDENT);
+            mHIRCache.put(key, BlockType.HIR_RESIDENT);
+            hirBytes += blocksize;
+          }
+          mBlockIdToSize.put(blockId, blocksize);
+        }
+        mSpaceManager.put(location, new SpaceContainer(hirBytes, lirBytes, totalBytes));

```

Comment:
In mSpaceManager, totalBytes is assumed to be a constant value, but if some blocks are pinned during this initialization and unpinned afterwards, then totalBytes will change. So it is reasonable to dynamically calculate the totalBytes when checking space limit of HIR and LIR cache.